velox/exec/tests/RowContainerTest.cpp

```diff
@@ -59,17 +68,59 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
     }
   }
 
+  void testExtractColumnOffsetsForOddRows(
+      RowContainer& container,
+      const std::vector<char*>& rows,
+      int column,
+      const VectorPtr& expected) {
+    auto size = rows.size();
+
+    // Extract only odd rows.
+    // Test using extractColumnOffsets API.
+    auto oddsSize = size / 2;
+    auto offsetsBuffer =
+        AlignedBuffer::allocate<vector_size_t>(oddsSize, pool_.get());
+    offsetsBuffer->setSize(oddsSize * sizeof(vector_size_t));
+    auto rawOffsetsBuffer = offsetsBuffer->asMutable<vector_size_t>();
+    for (int i = 0; i < oddsSize; i++) {
+      rawOffsetsBuffer[i] = i * 2;
+    }
+
+    auto result = BaseVector::create(expected->type(), oddsSize, pool_.get());
+    container.extractColumnAtOffsets(
```

Comment:
Would you add a test case where resultOffset is > 0 ?

Do we need a check somewhere that resultOffset + numRows < result->size()?

Reply:
I added a test case where resultOffset is > 0.

Had seen only DCHECKs and a single checkConsistency method with CHECKs so didn't any one. But have added one now.